mmr/src/lib.rs

```diff
@@ -21,6 +21,7 @@ const EMPTY_VALUE: [u8; 32] = [0; 32];
 #[derive(Debug, Clone)]
 pub struct MerkleMountainRange<T, Hash, const MAX_HEIGHT: u8 = 32> {
     roots: StackSync<Root>,
+    len: usize,
```

Comment:
Can you not derive len from roots?
e.g.

    roots: [Root{height:3}, Root{height:1}]

implies `2^(3-1) + 2^(1-1) = 4 + 1 = 5` elements in the MMR